test/OrchardCore.Tests/Modules/OrchardCore.ContentFields/Settings/DriverTests.cs

```diff
@@ -0,0 +1,277 @@
+                            TextPlaceholder = "Test Text Placeholder",
+                            DefaultUrl = "https://www.orchardcore.net",
+                            DefaultText = "Test Text",
+                            DefaultTarget = "Test Target",
+                        }));
+        var shapeResult = await GetShapeResult(contentDefinition, new LinkFieldSettingsDriver());
+        var settings = (LinkFieldSettings)shapeResult.Shape;
+
+        Assert.Equal("Test Hint", settings.Hint);
+        Assert.Equal("Test Hint Link Text", settings.HintLinkText);
+        Assert.True(settings.Required);
+        Assert.Equal(LinkTextMode.Static, settings.LinkTextMode);
+        Assert.Equal("Test Url Placeholder", settings.UrlPlaceholder);
+        Assert.Equal("Test Text Placeholder", settings.TextPlaceholder);
+        Assert.Equal("https://www.orchardcore.net", settings.DefaultUrl);
+        Assert.Equal("Test Text", settings.DefaultText);
+        Assert.Equal("Test Target", settings.DefaultTarget);
+    }
+
+    [Fact]
+    public async Task LocalizationSetContentPickerFieldSettingsShouldDeserialize()
+    {
+        var contentDefinition =
+            BuildContentPartWithField(field => field
+                        .OfType(nameof(LocalizationSetContentPickerField))
+                        .WithSettings(new LocalizationSetContentPickerFieldSettings
+                        { 
+                            Hint = "Test Hint",
+                            Required = true,
+                            Multiple = true,
+                            DisplayedContentTypes = ["one", "two", "three"],
+                        }));
+
+        var shapeResult = await GetShapeResult(contentDefinition, new LocalizationSetContentPickerFieldSettingsDriver());
+        var settings = (LocalizationSetContentPickerFieldSettings)shapeResult.Shape;
+
+        Assert.Equal("Test Hint", settings.Hint);
+        Assert.True(settings.Required);
+        Assert.True(settings.Multiple);
+        Assert.Equal<string[]>(["one", "two", "three"], settings.DisplayedContentTypes);
+    }
+
+    [Fact]
+    public async Task NumericFieldSettingsShouldDeserialize()
+    {
+        var contentDefinition =
+            BuildContentPartWithField(field => field
+                        .OfType(nameof(NumericField))
+                        .WithSettings(new NumericFieldSettings
+                        {
+                            Hint = "Test Hint",
+                            Required = true,
+                            Scale = 4,
+                            Minimum = 3,
+                            Maximum = 1000,
+                            Placeholder = "Test Placeholder",
+                            DefaultValue = "Test Default Value",
+                        }));
+
+        var shapeResult = await GetShapeResult(contentDefinition, new NumericFieldSettingsDriver());
+        var settings = (NumericFieldSettings)shapeResult.Shape;
```

Comment:
shouldn't you be getting the settings from the contentDefinition instead? `contentDefinition.GetSettings<NumericFieldSetting>()`

Then you'll no longer need the GetShapeResult

Reply:
If I get settings from the contentDefinition directly, then I think it bypasses the Driver.

The bug looks to be in the Driver implementation itself using the `Settings.ToObject` syntax instead of the `GetSettings` method. So I tried to design the test to exercise and test the Driver implementation.

Incidentally, this actually uncovered a [separate bug](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/YoutubeFieldSettingsDriver.cs#L16-L19) in the `YoutubeSettingsDriver` where the Hint, Label and Required assignments were missing, and the Width and Height assignments weren't using the settings. I went ahead and [fixed that as well](https://github.com/davidpuplava/OrchardCore/blob/field-settings-empty-fix/src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/YoutubeFieldSettingsDriver.cs#L18-L22).